src/main/kotlin/org/elm/ide/inspections/MissingCaseBranchAdder.kt

```diff
@@ -60,27 +61,32 @@ class MissingCaseBranchAdder(val element: ElmCaseOfExpr) {
         element.addRange(trailingWs.first(), trailingWs.last())
     }
 
-    private fun calcMissingBranches(): List<TyUnion.Variant> {
-        val inference = element.findInference() ?: return emptyList()
+    private fun calcMissingBranches(): VariantParameters {
+        val inference = element.findInference() ?: return emptyMap()
 
         // This only works on case expressions with a union type for now
         val exprTy = element.expression?.let { inference.elementType(it) } as? TyUnion
-                ?: return emptyList()
+                ?: return emptyMap()
 
-        val allBranches = exprTy.variants.associateBy { it.name }
+        val project = element.project
+        val declaration = ElmNamedElementIndex.find(exprTy.name, project, GlobalSearchScope.allScope(project))
```

Comment:
Feels like this should be moved to a more central location. Either closer to where `TyUnion` is defined or, if you don't want to dirty that up, add it as a convenience function on `ElmNamedElementIndex`.

Reply:
We can make this change down the road when the need arises.